calculator/README.md

```diff
@@ -0,0 +1,113 @@
+# Calculator
+
+_Due Monday, July 20th, before midnight Pacific Time._
+
+In this assignment you'll implement a simple calculator that understands Reverse
+Polish Notation (also known as postfix notation).  This is an alternative way to
+write mathematical expressions.  Each operator is written _after_  its operands.
+For example, the standard (infix) expression...
+
+```
+7 - 4
+```
+
+...would become:
+
+```
+7 4 -
+```
+
+And the more complicated equation...
+
+```
+((1 + 2) * (7 - 3)) ^ 2
+```
+
+...would become:
+
+```
+1 2 + 7 3 - * 2 ^
+```
+
+Reverse Polish Notation is much easier for computers to work with (notice how no
+parentheses  are required),  so many  old calculators used it.  In fact, You can
+think of Reverse Polish Notation  as a stream of instructions  for a very simple
+type of computer: a **stack machine**.
+
+
+## Stack Machines
+
+A stack machine is a  computer that stores its data  on a stack.  To evaluate an
+RPN expression, a stack machine starts with an empty stack.  It then reads input
+as a sequence of **tokens**, from left to right:
+
+- If the token is a number, the stack machine pushes it onto the stack.
+- If the token is an operator,  the stack machine pops numbers off of the stack,
+  performs the operation on those numbers, and pushes the result onto the stack.
+
+If evertything goes well, there will be one number on the stack after all tokens
+have been processed.  This is the final value of the input expression.
+
+
+## Your Assignment
+
+- Implement a stack that stores doubles.
+- Implement an RPN calculator using your stack class.
+- You can't use any container types from the standard library.
+- Make sure you don't print anything that isn't explicitly allowed.
+- Make sure you don't have any memory leaks.
+
+
+## The Stack
+
+The  `Stack.h`  file describes an  abstract  Stack  data structure.  Implement a
+concrete Stack in  `MyStack.h` and `MyStack.cpp`.  You'll probably want to use a
+vector or linked list as the underlying structure.
```

Comment:
Just to be clear, when you say use a vector, you are __not__ referring to `std::vector` right? Because the previous para states that you can't use any container types from the standard library.

Reply:
Yep.  It has to be a custom vector implementation.  Using a `std::list` as your linked list would be illegal as well.